### PR TITLE
remove some warnings

### PR DESCRIPTION
**Pull request by EdwardLi-coder**
When I run the tests, I often encounter warnings related to library dependencies. In my PR, I've ignored warnings coming from third-party libs. And if they come from our own code, I'm rewriting it.
These warnings are need to rewrite code.
```
tests/unit/lib/test_text.py: 1 warning
  /Users/lifei/PycharmProjects/datachain/.nox/tests-3-12/lib/python3.12/site-packages/datachain/lib/text.py:36: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
    tokens = torch.tensor(tokens)

tests/unit/lib/test_clip.py::test_similarity_scores_hf
tests/unit/lib/test_image.py::test_convert_image_hf
  /Users/lifei/PycharmProjects/datachain/.nox/tests-3-12/lib/python3.12/site-packages/datachain/lib/image.py:37: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
    img = torch.tensor(img.pixel_values[0])  # type: ignore[assignment,attr-defined]

tests/unit/lib/test_clip.py::test_similarity_scores_hf
  /Users/lifei/PycharmProjects/datachain/.nox/tests-3-12/lib/python3.12/site-packages/datachain/lib/clip.py:21: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
    return lambda x: method(torch.tensor(x))
```

### CODE REVIEW

**Comment by shcheklein** [comment]
can we ignore warning that come from third-party libs (e.g. some lib that we depend on is using torch in a deprecated way) and fix warnings on our end by rewriting the code, wdyt?

**Comment by EdwardLi-coder** [comment]
> can we ignore warning that come from third-party libs (e.g. some lib that we depend on is using torch in a deprecated way) and fix warnings on our end by rewriting the code, wdyt?

You can review my code. I have already done this.

**Comment by shcheklein** [comment]
> In my PR, I've ignored warnings coming from third-party libs. And if they come from our own code, I'm rewriting it.

will it ignore new warnings (let's say someone adds it by mistake tomorrow) in our code? just to double check ...

**Comment by EdwardLi-coder** [comment]
> > In my PR, I've ignored warnings coming from third-party libs. And if they come from our own code, I'm rewriting it.
> 
> will it ignore new warnings (let's say someone adds it by mistake tomorrow) in our code? just to double check ...

These ignore settings are specific to warnings from certain third-party libraries. They will not affect new warnings in our own code. Each ignore has a specific module path, so even if someone accidentally adds new warnings in our code in the future, these settings will not ignore them. New warnings in our code will still be captured and displayed.
However, to ensure we don't miss important warnings, I suggest we periodically review these ignore settings and re-evaluate their necessity after updates to the third-party libraries.

File: pyproject.toml [hunk 1]
```diff
@@ -141,6 +141,10 @@ filterwarnings = [
   "error::pytest_mock.PytestMockWarning",
   "error::pytest.PytestCollectionWarning",
   "error::sqlalchemy.exc.SADeprecationWarning",
+  "ignore::DeprecationWarning:timm.*",
+  "ignore::DeprecationWarning:botocore.auth",
+  "ignore::DeprecationWarning:datasets.utils._dill",
```
Comment (EdwardLi-coder) [inline pyproject.toml:146]: I think maybe only ignore DeprecationWarning is good.

File: src/datachain/lib/clip.py [hunk 2]
```diff
@@ -18,7 +18,7 @@ def _get_encoder(model: Any, type: Literal["image", "text"]) -> Callable:
         hasattr(model, method_name) and inspect.ismethod(getattr(model, method_name))
     ):
         method = getattr(model, method_name)
-        return lambda x: method(torch.tensor(x))
+        return lambda x: method(torch.as_tensor(x).clone().detach())
```
Comment (EdwardLi-coder) [inline src/datachain/lib/clip.py:21]: This is the warning:
```
tests/unit/lib/test_clip.py::test_similarity_scores_hf
  /Users/lifei/PycharmProjects/datachain/.nox/tests-3-12/lib/python3.12/site-packages/datachain/lib/clip.py:21: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
    return lambda x: method(torch.tensor(x))
```
Update Reason:
We use torch.as_tensor() to create tensors, which avoids copying data whenever possible. We then call clone().detach() to create a new tensor that is detached from the original data and will not affect the computation graph of the original data.

File: src/datachain/lib/image.py [hunk 3]
```diff
@@ -34,7 +34,7 @@ def convert_image(
             from transformers.image_processing_utils import BaseImageProcessor
 
             if isinstance(transform, BaseImageProcessor):
-                img = torch.tensor(img.pixel_values[0])  # type: ignore[assignment,attr-defined]
+                img = torch.as_tensor(img.pixel_values[0]).clone().detach()  # type: ignore[assignment,attr-defined]
```
Comment (EdwardLi-coder) [inline src/datachain/lib/image.py:37]: This is the warning:
```
tests/unit/lib/test_clip.py::test_similarity_scores_hf
tests/unit/lib/test_image.py::test_convert_image_hf
  /Users/lifei/PycharmProjects/datachain/.nox/tests-3-12/lib/python3.12/site-packages/datachain/lib/image.py:37: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
    img = torch.tensor(img.pixel_values[0])  # type: ignore[assignment,attr-defined]
```
Update Reason:
We use torch.as_tensor() to create tensors, which avoids copying data whenever possible. We then call clone().detach() to create a new tensor that is detached from the original data and will not affect the computation graph of the original data.

**Comment by EdwardLi-coder** [comment]
@shcheklein I only remain ignore the DeprecationWarning. I have provided some explanations for my modifications. If you have any other suggestions, I would be happy to make further changes.

**Review by dreadatour — COMMENTED** [review]
Looks good to me! 👍

**Review by shcheklein — APPROVED** [review]

**Comment by shcheklein** [comment]
thanks @EdwardLi-coder !
